stable/database/files/nuobackup

```diff
@@ -0,0 +1,161 @@
+        "--type="* )
+           backup_type="${opt#*=}";;
+        "--db-name" )
+           db_name="$1"; shift;;
+        "--db-name="* )
+           db_name="${opt#*=}";;
+        "--group" )
+           backup_group="$1"; shift;;
+        "--group="* )
+           backup_group="${opt#*=}";;
+        "--labels" )
+           labels="$1"; shift;;
+        "--labels="* )
+           labels="${opt#*=}";;
+        "--semaphore" )
+           semaphore="$1"; shift;;
+        "--semaphore="* )
+           semaphore="${opt#*=}";;
+        "--timeout" )
+           timeout="$1"; shift;;
+        "--timeout="* )
+           timeout="${opt#*=}";;
+        "--backup-root" )
+           backup_root="$1"; shift;;
+        "--backup-root="* )
+           backup_root="${opt#*=}";;
+        *) echo >&2 "Invalid nuobackup option: $opt ($@)"; exit 1;;
+   esac
+done
+
+backup_type=$(echo $backup_type | tr '[:upper:]' '[:lower:]')
+label=$(echo $label | tr '[:upper:]' '[:lower:]')
+
+# Return which group made the latest backup
+if [ "$backup_type" = "report-latest" -a "$backup_group" = "" ]; then
+   echo "$(nuocmd --api-server $NUOCMD_API_SERVER get value --key $NUODB_BACKUP_KEY/$db_name/latest )"
+   exit 0
+fi
+
+# if $semaphore is not null, then check the value in Raft
+if [ -n "$semaphore" ]; then
+   if [ "$backup_group" != "" ]; then
+      gate=$(nuocmd --api-server $NUOCMD_API_SERVER get value --key $semaphore/$db_name/$backup_group)
+
+      # and reset the semaphore
+      nuocmd --api-server $NUOCMD_API_SERVER set value --key $semaphore/$db_name/$backup_group --value '' --expected-value "$gate"
+   else
+      gate=$(nuocmd --api-server $NUOCMD_API_SERVER get value --key $semaphore/$db_name)
+
+      # and reset the semaphore
+      nuocmd --api-server $NUOCMD_API_SERVER set value --key $semaphore/$db_name --value '' --expected-value "$gate"
+   fi
+
+
+   if [ -z "$gate" ]; then
+      echo >&2 "Semaphore $semaphore/$db_name empty - nothing to do"
+      exit 0
+   fi
+
+   # wait for the database to be RUNNING
+   nuocmd --api-server $NUOCMD_API_SERVER check database --db-name $db_name --check-running --wait-forever
+fi
+
+# find the index for the latest backup info
+latest=$(nuocmd --api-server $NUOCMD_API_SERVER get value --key $NUODB_BACKUP_KEY/$db_name/${backup_group}/latest )
+current=$latest
+
+if [ "$backup_type" = "full" ]; then
+   if [ -z "$latest" ]; then
+      current=0
+   fi
+
+   backupset=$( date "+%Y%m%dT%H%M%S" )
+
+   # bump latest, and wrap around (simple ring-buffer)
+   next=$(( (current + 1) % $NUODB_MAX_BACKUP_HISTORY ))
+else
+   if [ -z "$latest" ]; then
+      echo >&2 "No full backup has completed yet - try again later"
+      exit -1
+   fi
+
+   # retrieve the latest backupset name
+   backupset=$(nuocmd --api-server $NUOCMD_API_SERVER get value --key $NUODB_BACKUP_KEY/$db_name/${backup_group}/${latest} )
+   echo >&2 "Looking up $NUODB_BACKUP_KEY/$db_name/${backup_group}/${latest} returned: $backupset"
```

Comment:
This looks like INFO message. Why it is redirected to stderr?

Reply:
The caller reads stdout to get the result of a `get-latest` operation, so we should avoid sending diagnostic information to stdout.